### ISSUE TITLE
cannot create composite key in mysql with automigrate

### ISSUE THREAD

**Issue by nkovacs** [jinzhu/gorm#601]
Automigrating this struct:

```go
type AuthAssignment struct {
	Group string `gorm:"primary_key"`
	UserID uint64 `gorm:"primary_key"`
}
```

results in this sql code:

```SQL
CREATE TABLE `auth_assignments` (`group` varchar(255),`user_id` bigint unsigned AUTO_INCREMENT , PRIMARY KEY (`group`,`user_id`)) 
```

Which doesn't work: Incorrect table definition; there can be only one auto column and it must be defined as a key 

AUTO_INCREMENT should not be added to a composite key's integer field(s).

**Comment by mrkaspa** [comment]
+1 having the same error

**Comment by jinzhu** [comment]
set type with tag `sql:"type=integer" or use unique index

**Comment by nkovacs** [comment]
I set the type to bigint, but now, months later, when I redid the migration, it failed because the uint64 primary key became bigint unsigned (a change in gorm's code I think), and I couldn't create the foreign key because of mismatched types. Not having to explicitly specify the type also makes the code more portable (bigint unsigned wouldn't work in pgsql), since gorm will (hopefully) use the same sql type for both fields in both tables.
Fortunately, since I first encountered this issue, you've added the option to disable autoincrement using the `sql:AUTO_INCREMENT:FALSE` tag, so I'm going to use that.
